utils/create-module.js

```diff
@@ -0,0 +1,131 @@
+  }
+};
+
+/**
+ * Add export * from '@workday/canvas-kit-react-<MODULE_NAME>' to
+ * modules/_canvas-kit-react/index.ts so that consumers can use.
+ */
+const addExportToIndex = (cwd, moduleName, wantsCss) => {
+  const sortExports = lines => {
+    const exportsAsObj = lines
+      .filter(line => !line.includes("from '@workday/canvas-kit-react-core'"))
+      // Ignore last line in file :)
+      .filter(line => line.length > 0)
+      .reduce((acc, line) => {
+        // Split out module name and slice off trailing '; part
+        const modName = line.split('canvas-kit-react-')[1].slice(0, -2);
+        acc[modName] = line;
+        return acc;
+      }, {});
+    const sorted = Object.keys(exportsAsObj)
+      .sort()
+      .reduce((acc, key) => {
+        acc.push(exportsAsObj[key]);
+        return acc;
+      }, []);
+    sorted.push(''); // Will result in EOF line
+    return sorted;
+  };
+
+  const indexPath = path.join(cwd, 'modules/_canvas-kit-react/index.ts');
+  const lines = fs.readFileSync(indexPath, 'utf8').split('\n');
+
+  // Add our module's export statement then sort
+  const exportStatement = `export * from '@workday/canvas-kit-react-${moduleName}';`;
+  lines.push(exportStatement);
+  const sortedExports = sortExports(lines);
+
+  // Place our canvas kit export at the top
+  sortedExports.unshift("export {default as canvas} from '@workday/canvas-kit-react-core';");
+  fs.writeFileSync(indexPath, sortedExports.join('\n'), 'utf8');
+};
+
+const addImportToSassIndex = (cwd, moduleName) => {
+  const sortImports = lines => {
+    const importsAsObj = lines
+      .filter(line => !line.includes('$wdc-system-icons-path'))
+      .filter(line => line.length > 0)
+      .reduce((acc, line) => {
+        // Split out module name and slice off trailing /index.scss';
+        const modName = line.split('canvas-kit-css-')[1].split("/index.scss';")[0];
+        acc[modName] = line;
+        return acc;
+      }, {});
+    const sorted = Object.keys(importsAsObj)
+      .sort()
+      .reduce((acc, key) => {
+        acc.push(importsAsObj[key]);
+        return acc;
+      }, []);
+    sorted.push(''); // Will result in EOF line
+    return sorted;
+  };
+
+  const indexPath = path.join(cwd, 'modules/_canvas-kit-css/index.scss');
+  const lines = fs.readFileSync(indexPath, 'utf8').split('\n');
+
+  // Add our module's import statement then sort
+  const importStatement = `@import '~@workday/canvas-kit-css-${moduleName}/index.scss';`;
+  lines.push(importStatement);
+  const sortedImports = sortImports(lines);
+
+  // Place our system icons path at the top
+  sortedImports.unshift(
+    "$wdc-system-icons-path: '../../node_modules/@workday/canvas-system-icons-web/dist/svg' !default;"
+  );
+  fs.writeFileSync(indexPath, sortedImports.join('\n'), 'utf8');
+};
+
+// Only allow this script to be ran off command line
+if (require.main === module) {
+  const cwd = process.cwd();
+  const moduleName = process.argv[2];
```

Comment:
The following 8 lines are a lot of logic to parse. I find the following much easier to understand:

```ts
const [_, __, moduleName, moduleType] = process.argv;
const wantsCss = moduleType.toLowerCase() === 'css'
```

And a friendlier error-handling version:
```ts
const [_, __, moduleName, moduleType = 'react'] = process.argv

if (!['css', 'react'].includes(moduleType)) {
  throw Error(`Module type must be either 'react' or 'css'`)
}
```

Going a little further, a boolean argument usually indicates you have 2 separate functions. Refactoring a bit:
```ts
if (require.main === module) {
  const cwd = process.cwd();
  const [_, __, moduleName, moduleType = 'react'] = process.argv

  if (!['css', 'react'].includes(moduleType)) {
    throw Error(`Module type (second argument) must be either 'react' or 'css'`)
  }

  addModuleAsDependencyByType(cwd, moduleName, moduleType);
  addExportToIndexByType(cwd, moduleName, moduleType);
}
```

Reply:
I really like the destructuring approach you've outlined, I'll incorporate it for sure!

I'm thinking we not need the error handling as the 'css' is optional, but the 'react' step will always happen. Also, I think it would never fail...

If I'm not misunderstanding:
```js
  const [_, __, moduleName, moduleType = 'react'] = process.argv
```
Is an ES6 destructuring with also a default parameter being set for `moduleType` such that if `process.argv` is falsy at `[3]`, then `moduleType === 'react'` will always be true, correct? If so,

```js
 !['css', 'react'].includes(moduleType)
```
will never fail no?

If agreed, I'll just refactor to use pretty much everything in your example less the error handling. Sound good? LMK, thanks Nicholas!

Reply:
My thought was changing what you have in `create-module.sh` to call `create-module.js` multiple times - once for 'css' and once for 'react'. In the case you only want react, call it with 'react'. If you want both, call it for 'react' and another to call it with 'css'

> I'm thinking we not need the error handling as the 'css' is optional, but the 'react' step will always happen. Also, I think it would never fail...

The following will fail:
```
node ./utils/create-module.js myModule angular
```

The only commands that will pass this check are as follows:
```
node ./utils/create-module.js myModule
node ./utils/create-module.js myModule react
node ./utils/create-module.js myModule css
```

Personally I'd get rid of the default and require `react` and `css`

Reply:
Gotcha, thanks for clarification 👍 
> require react and css
Makes sense

Reply:
Thanks for guidance @NicholasBoll ... I believe this is now working for the nominal cases. 

Running the node script directly without 'react' or 'css' works as intended:

![image](https://user-images.githubusercontent.com/142403/61984276-95962880-afb8-11e9-852c-673f334d993a.png)